src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java

```diff
@@ -240,28 +239,23 @@ private Mono<List<CommodityTO>> createCommoditiesByBookingIDAndTOs(
         .collectList();
   }
 
-  private Mono<List<ValueAddedServiceRequestTO>>
-      createValueAddedServiceRequestsByBookingIDAndTOs(
-          UUID bookingID, List<ValueAddedServiceRequestTO> valueAddedServiceRequests) {
+  private Mono<List<ValueAddedServiceRequestTO>> createValueAddedServiceRequestsByBookingIDAndTOs(
+      UUID bookingID, List<ValueAddedServiceRequestTO> valueAddedServiceRequests) {
 
     if (Objects.isNull(valueAddedServiceRequests) || valueAddedServiceRequests.isEmpty()) {
       return Mono.empty();
     }
 
-    Stream<ValueAddedServiceRequest> vasrStream =
-        valueAddedServiceRequests.stream()
-            .map(
-                vasr -> {
-                  ValueAddedServiceRequest valueAddedServiceRequest =
-                      new ValueAddedServiceRequest();
-                  valueAddedServiceRequest.setBookingID(bookingID);
-                  valueAddedServiceRequest.setValueAddedServiceCode(
-                      vasr.getValueAddedServiceCode());
-                  return valueAddedServiceRequest;
-                });
-
-    return valueAddedServiceRequestRepository
-        .saveAll(Flux.fromStream(vasrStream))
+    return Flux.fromIterable(valueAddedServiceRequests)
+        .map(
+            valueAddedServiceRequestTO -> {
+              ValueAddedServiceRequest valueAddedServiceRequest = new ValueAddedServiceRequest();
+              valueAddedServiceRequest.setBookingID(bookingID);
+              valueAddedServiceRequest.setValueAddedServiceCode(
+                  valueAddedServiceRequestTO.getValueAddedServiceCode());
+              return valueAddedServiceRequest;
+            })
+        .flatMap(valueAddedServiceRequestRepository::save)
```

Comment:
I would probably use `.concatMap` here to preserve ordering.  It probably only matters if this is also echo'ed in the output though.

Reply:
changed accordingly